core/runtime/execute_engine.cpp

```diff
@@ -121,36 +121,30 @@ std::vector<at::Tensor> execute_engine(std::vector<at::Tensor> inputs, c10::intr
     }
   }
 
-  std::vector<void*> gpu_handles;
-  std::vector<at::Tensor> contig_inputs{};
   {
     std::unique_ptr<torch::autograd::profiler::RecordProfile> input_profiler_guard;
     if (compiled_engine->profile_execution) {
       input_profiler_guard =
           std::make_unique<torch::autograd::profiler::RecordProfile>(compiled_engine->input_profile_path);
     }
-
-    contig_inputs.reserve(inputs.size());
-
     for (size_t i = 0; i < inputs.size(); i++) {
-      uint64_t pyt_idx = compiled_engine->in_binding_map[i];
+      std::string name = compiled_engine->exec_ctx->getEngine().getIOTensorName(i);
```

Comment:
@peri044 why arent we using input_binding_names here. TensorRT does not guarantee binding order. This would be potentially unsafe. 

I would expect that the workflow here would be to iterate not through `inputs.size()` but through the names in `input_binding_names`

Reply:
this should be `std::string name = compiled_engine->input_binding_names[i]`